### ISSUE TITLE
"dead code" warning for short circuiting isn't complete

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/sdk#29196]
In trivial cases its pretty easy to see what atually is/isn't dead code, in part because of the partial warning. But in more complicated cases, it may create confusion that the "dead code" section does not highlight all of the dead code (in mixtures of dead and live code, etc).

repro.dart:
```
main() {                                                                         
  return false && 1 == 2 && 3 == 4;                                              
}
```

What happens:

`1 == 2` is highlighted as dead code because its after `false &&` so won't do anything.

What I expect to happen:

`1 == 2 && 3 == 4` is highlighted, because that entire section is dead, not just the `1 == 2`.

**Comment by srawlins** [comment]
Yeah the dead code checking in binary logical expressions is very limited. In particular, the [`_getConstantBooleanValue`](https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/src/generated/resolver.dart#L2407) method has a big chunk of text commented out with no explanation other than:

> Don't consider situations where we could evaluate to a constant boolean
> expression with the ConstantVisitor

But why not? Performance? I'm not sure that the/a ConstantVisitor would reconcile this, but I shouldn't think it would be hard...

**Comment by srawlins** [comment]
In addition to number literals, boolean literals should also be included in this request. Basically https://github.com/dart-lang/sdk/issues/3511:

```dart
while (true) {
  if (false) break;
}
```

**Comment by srawlins** [comment]
The correct range is now highlighted.
